File: src/2848.points-that-intersect-with-cars.3034/solution.ts
function numberOfPoints (nums: number[][]): number {
    const counts = new Array<number>(102).fill(0);
    for (const [start, end, ] of nums) {
        counts[start]++;
        counts[end + 1]--;
    }
    let result = 0;
    for (let i = 1; i < 101; i++) {
        counts[i] += counts[i - 1];
        if (counts[i] !== 0) {
            result++;
        }
    }
    return result;
}

File: src/2848.points-that-intersect-with-cars.3034/solution.java
class Solution {
    public int numberOfPoints(List<List<Integer>> nums) {
        int[] parks = new int[102];
        nums.forEach((list)->{
            parks[list.getFirst()]++;
            parks[list.get(1)+1]--;
        });

        int result = 0;
        for(int i=1;i<parks.length;i++){
            parks[i] += parks[i-1];
            if(parks[i]>0){
                result++;
            }
        }

        return result;
                
    }
}